packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,865 @@
+    pendingVote.value  = pendingVote.value.add(value);
+    pendingVote.blockNumber = block.number;
+  }
+
+  /**
+   * @notice Decrements the number of pending votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function decrementPendingVotes(address group, address account, uint256 value) private {
+    PendingVotes storage pending = votes.pending;
+    pending.total = pending.total.sub(value);
+
+    GroupPendingVotes storage groupPending = pending.forGroup[group];
+    groupPending.total = groupPending.total.sub(value);
+
+    PendingVote storage pendingVote = groupPending.byAccount[account];
+    pendingVote.value = pendingVote.value.sub(value);
+    if (pendingVote.value == 0) {
+      pendingVote.blockNumber = 0;
+    }
+  }
+
+  /**
+   * @notice Increments the number of active votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function incrementActiveVotes(address group, address account, uint256 value) private {
+    ActiveVotes storage active = votes.active;
+    active.total = active.total.add(value);
+
+    uint256 unitsDelta = getActiveVotesUnitsDelta(group, value);
+
+    GroupActiveVotes storage groupActive = active.forGroup[group];
+    groupActive.total = groupActive.total.add(value);
+
+    groupActive.totalUnits = groupActive.totalUnits.add(unitsDelta);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].add(unitsDelta);
+  }
+
+  /**
+   * @notice Decrements the number of active votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function decrementActiveVotes(address group, address account, uint256 value) private {
+    ActiveVotes storage active = votes.active;
+    active.total = active.total.sub(value);
+
+    uint256 unitsDelta = getActiveVotesUnitsDelta(group, value);
+
+    GroupActiveVotes storage groupActive = active.forGroup[group];
+    groupActive.total = groupActive.total.sub(value);
+
+    groupActive.totalUnits = groupActive.totalUnits.sub(unitsDelta);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(unitsDelta);
+  }
+
+  /**
+   * @notice Returns the delta in active vote denominator for `group`.
+   * @param group The address of the validator group.
+   * @param value The number of active votes being added.
+   * @return The delta in active vote denominator for `group`.
+   */
+  function getActiveVotesUnitsDelta(address group, uint256 value) private view returns (uint256) {
+    // Preserve unitsDelta * total = value * totalUnits
+    return value.mul(votes.active.forGroup[group].totalUnits.add(1)).div(
```

Comment:
Not sure to follow the math here.
So the relation is:
```
accountUnits / totalUnits = accountValue / totalValue
```
where value = votes

The question is: how much units would i gain if a add `X` value/votes

that means i need to solve this equation:
```
(newUnits) / (totalUnits + newUnits) = (addedValue) / (totalValue + addedValue)
```
where we need to get `newUnits` and the rest is given.

which doesn't look similar to what we are doing here... plus we are adding `1` to both `unit` and `totalValue`. And is not like we are (a) creating only one unit. (b) adding only 1 to totalValue.

Reply:
The equation you wrote simplifies to `totalUnits * addedValue = newUnits * totalValue`, which, solving for `newUnits`, gives us `totalUnits * addedValue / totalValue`, i.e. the equation in the code.

If we don't add 1 to those values no one can ever place a vote, since the numerator and denominator will both be zero

Reply:
This is much better, thanks!